### ISSUE TITLE
How do I define my own plug?

### ISSUE THREAD

**Issue by aronasorman** [phoenixframework/phoenix#465]
I've been trying to create my own `:api` pipeline, which basically just makes `json` the default template to use. I defined the following in my `router.ex`:

``` elixir
  pipeline :api do
    plug :json_api
  end
```

And it seems that it's just looking for a function in the same module named `json_api`, so I just defined that:

``` elixir
  # custom plugs
  def json_api(conn, _opts) do
    conn
    |> put_resp_content_type("application/json")
  end
```

However, even after putting my controllers inside a scope and inserting the `pipe_through :api` line inside the scope block, my controller is still looking for a `.html` template rather than the json template I have. Putting in `put_resp_content_type` manually inside the controller function makes it search for the `.json` instead. 

So basically I'm not entirely sure how the whole plug pipeline works, or what I'm missing here. Any help in here would be great.
